### PR TITLE
Feature/activity summary

### LINKED ISSUE #36: Add Status to ActivitySummary
**Is your feature request related to a problem? Please describe.**
The ActivitySummary class needs to refer to draft and completed versions in one list

**Describe the solution you'd like**
[ActivitySummary](https://github.com/Kaleido-Biosciences/fetch/blob/develop/src/main/java/com/kaleido/fetch/domain/ActivitySummary.java) should change the attribute **completedVersions** to **versions** and update any references.

**Additional context**
We'll also need an Endpoint to retrieve the activity summaries from Cabinet.  It needs to combine information from the kapture client experiment as well as the information from the endpoint this ticket will create.
https://github.com/Kaleido-Biosciences/cabinet/issues/10

### PR DESCRIPTION
**Pull request by QParavin**
1. Added the null condition to getAllBarcodes service
2. Added the endpoint for activitySummary including the activity version
3. Formatted the code

### CODE REVIEW

File: src/main/java/com/kaleido/fetch/service/ActivityService.java [hunk 1]
```diff
@@ -157,9 +159,17 @@ public Activity getActivity(Long id) {
         return restTemplate.exchange(plateMapURI, HttpMethod.POST, entity, PlateMap[].class);
     	
     }
-
-    private List<Experiment> searchExperiment(String searchTerm) {
-    	
+    
+    public List<ActivitySummary> getActivitySummaryList(String searchTerm) {
+  	    log.info("ActivitySummary name is ", searchTerm);
```
Comment (wfowlks) [inline src/main/java/com/kaleido/fetch/service/ActivityService.java]:
```suggestion
  	    log.debug("ActivitySummary name is ", searchTerm);
```

File: src/main/java/com/kaleido/fetch/service/ActivityService.java [hunk 2]
```diff
@@ -157,9 +159,17 @@ public Activity getActivity(Long id) {
...
+  	    List<ActivitySummary> activitySummaryList = this.findActivities(searchTerm).stream().map(this::buildActivitySummary).collect(Collectors.toList());
```
Comment (wfowlks) [inline src/main/java/com/kaleido/fetch/service/ActivityService.java]:
```suggestion
  	    List<ActivitySummary> activitySummaryList = this.findActivities(searchTerm).stream()
  	        .map(this::buildActivitySummary)
  	        .collect(Collectors.toList());
```

File: src/main/java/com/kaleido/fetch/service/ActivityService.java [hunk 3]
```diff
@@ -157,9 +159,17 @@ public Activity getActivity(Long id) {
...
+  	    activitySummaryList.forEach(action ->{
+  		    action.getVersions().addAll(getPlateMapSummaryFromCabinet(action.getName()).stream().map(this::buildActivityVersion).collect(Collectors.toList()));
```
Comment (wfowlks) [inline src/main/java/com/kaleido/fetch/service/ActivityService.java]:
```suggestion
  		    action.getVersions().addAll(getPlateMapSummaryFromCabinet(action.getName()).stream()
  		        .map(this::buildActivityVersion)
  		        .collect(Collectors.toList()));
```

File: src/main/java/com/kaleido/fetch/service/ActivityService.java [hunk 4]
```diff
@@ -177,5 +187,34 @@ private Activity buildActivity(Experiment experiment) {
                 .description(experiment.getDescription())
                 .build();
     }
+   
+    private List<PlateMap> getPlateMapSummaryFromCabinet(String activityName) {
+        RestTemplate restTemplate = new RestTemplate();
+   	    String cabinetplateInfoURI = cabinetURI + "plate-map-summary/"+activityName;
```
Comment (wfowlks) [inline src/main/java/com/kaleido/fetch/service/ActivityService.java]: Please fix the indentation

File: src/main/java/com/kaleido/fetch/service/ActivityService.java [hunk 5]
```diff
@@ -177,5 +187,34 @@ private Activity buildActivity(Experiment experiment) {
...
+   	    ResponseEntity<List<PlateMap>> plateMapResponse =restTemplate.exchange(cabinetplateInfoURI,
```
Comment (wfowlks) [inline src/main/java/com/kaleido/fetch/service/ActivityService.java]: This is fine for now but we will probably want to create another ticket to refactor this to use the cabinet client

**Review by wfowlks — CHANGES_REQUESTED** [review]

File: src/main/java/com/kaleido/fetch/service/ActivityService.java (hunk 5 above)
Comment (QParavin) [reply src/main/java/com/kaleido/fetch/service/ActivityService.java]: create a ticket , we will work on Cabinet client

Comment (QParavin) [reply src/main/java/com/kaleido/fetch/service/ActivityService.java]: I updated direct call to cabinet with cabinet client

File: src/main/java/com/kaleido/fetch/service/ActivityService.java [hunk 6]
```diff
@@ -46,8 +50,8 @@
     private String cabinetURI;
 
     public ActivityService(KaptureClient<Experiment> experimentKaptureClient, CabinetClient<PlateMap> cabinetClient) {
-		this.cabinetClient = cabinetClient;
-		this.experimentKaptureClient = experimentKaptureClient;
+		    this.cabinetClient = cabinetClient;
```
Comment (wfowlks) [inline src/main/java/com/kaleido/fetch/service/ActivityService.java]: Can you do a find and replace of tab with space, it appears that there is a tab in the line before the spaces that you added.

File: src/main/java/com/kaleido/fetch/service/FetchService.java [hunk 7]
```diff
@@ -152,19 +152,19 @@ public FetchService(KaptureClient <Media> mediaKaptureClient, KaptureClient <Bat
...
-                platmapsfromrepository.getBody().stream()
-                    .forEach(plate -> {
-                        barcodeArray.add(plate.getBarcode());
-                    });
-            }
+        if(null!=experimentList.getBody() && !experimentList.getBody().isEmpty()) {
+        	var expertimentId = experimentList.getBody().get(0).getId().toString();
+        	if (experimentList.getStatusCode().is2xxSuccessful() && experimentList.getBody() != null && experimentList.getBody().size() > 0) {
+        		var platmapsfromrepository = plateMapKaptureClient.findByFieldEquals("experimentId", expertimentId);
+        		if (platmapsfromrepository.getStatusCode().is2xxSuccessful() && platmapsfromrepository.getBody() != null) {
```
Comment (wfowlks) [inline src/main/java/com/kaleido/fetch/service/FetchService.java]: Please do a find and replace of tabs with 4 spaces

**Review by wfowlks — CHANGES_REQUESTED** [review]
Pretty good, just a formatting issues and we should be good to merge this

File: src/main/java/com/kaleido/fetch/service/ActivityService.java (hunk 6 above)
Comment (QParavin) [reply src/main/java/com/kaleido/fetch/service/ActivityService.java]: Replaced tab with spaces

File: src/main/java/com/kaleido/fetch/service/FetchService.java (hunk 7 above)
Comment (QParavin) [reply src/main/java/com/kaleido/fetch/service/FetchService.java]: Replaced tab with spaces

**Review by wfowlks — APPROVED** [review]

**Pull request merged**